Sources/Rendering/OpenGL/Texture/index.js

```diff
@@ -613,15 +613,18 @@ function vtkOpenGLTexture(publicAPI, model) {
   };
 
   //----------------------------------------------------------------------------
-  function updateArrayDataType(dataType, data) {
+  function updateArrayDataType(dataType, data, depth = false) {
     const pixData = [];
     // if the opengl data type is float
     // then the data array must be float
     if (
       dataType !== VtkDataTypes.FLOAT &&
       model.openGLDataType === model.context.FLOAT
     ) {
-      const pixCount = model.width * model.height * model.components;
+      let pixCount = model.width * model.height * model.components;
+      if (depth) {
+        pixCount *= model.depth;
+      }
```

Comment:
We have three copies of the pixCount calc (my fault I suspect) Can we just pull that out and have one copy near the top of the method?

Reply:
I changed them